src/lib.rs

```diff
@@ -1,43 +1,52 @@
 use std::time::Duration;
 
-use ethers_core::{
-    types::{
-        transaction::eip2930::{AccessList, AccessListItem},
-        OtherFields, Transaction as EthersTx, H160, H256, U256, U64,
-    },
-    utils::rlp::{Decodable, Rlp},
+use alloy_rpc_engine_types::{
+    ExecutionPayload, ExecutionPayloadV1, ExecutionPayloadV2, ExecutionPayloadV3,
 };
+use ethereum_consensus::{ssz::prelude::deserialize, types::mainnet::SignedBeaconBlock};
+use reth_primitives::{Address, Bytes, TransactionSigned, TransactionSignedEcRecovered};
+use ssz::Decode;
 use tokio::sync::{mpsc, oneshot};
-use tokio_stream::{wrappers::UnboundedReceiverStream, StreamExt};
+use tokio_stream::{wrappers::UnboundedReceiverStream, Stream, StreamExt};
 use tonic::{codec::CompressionEncoding, transport::Channel, Request};
 
-pub mod api;
-pub mod eth;
-pub mod filter;
-pub mod types;
-
-use api::{
-    api_client::ApiClient, BlockSubmissionMsg, BlockSubmissionResponse, RawTxMsg, RawTxSequenceMsg,
-    TransactionResponse, TxFilter, TxSequenceMsg, TxSequenceResponse,
+pub mod generated;
+use generated::api::{
+    api_client::ApiClient, BlockSubmissionMsg, BlockSubmissionResponse, TransactionMsg,
+    TransactionResponse, TxFilter, TxSequenceMsgV2, TxSequenceResponse,
 };
-use eth::{CompactBeaconBlock, ExecutionPayload, ExecutionPayloadHeader, Transaction};
+
+const CLIENT_VERSION: &str = env!("CARGO_PKG_VERSION");
+const CLIENT_NAME: &str = env!("CARGO_PKG_NAME");
+
+/// Appends the api key metadata to the request, keyed by x-api-key.
+fn append_api_key<T>(req: &mut Request<T>, key: &str) {
+    req.metadata_mut().append("x-api-key", key.parse().unwrap());
+}
+
+/// Appends the client version metadata to the request, keyed by x-client-version.
+fn append_client_version<T>(req: &mut Request<T>) {
+    let client_full_name = format!("{}-rs-v{}", CLIENT_NAME, CLIENT_VERSION);
```

Comment:
```suggestion
    let client_full_name = format!("{}-rs/v{}", CLIENT_NAME, CLIENT_VERSION);
```

Just to follow conventions